console/src/view/task.rs

```diff
@@ -123,13 +163,134 @@ impl TaskView {
         let mut fields = Text::default();
         fields.extend(task.formatted_fields().iter().cloned().map(Spans::from));
 
+        // Bit of a deadlock: We cannot know the highest bucket value without determining the number of buckets,
+        // and we cannot determine the number of buckets without knowing the width of the chart area which depends on
+        // the number of digits in the highest bucket value.
+        // So just assume here the number of digits in the highest bucket value is 3.
+        // If we overshoot, there will be empty columns/buckets at the right end of the chart.
+        // If we undershoot, the rightmost 1-2 columns/buckets will be hidden.
+        // We could get the max bucket value from the previous render though...
+        let (chart_data, metadata) =
+            make_chart_data(self.details.clone(), task.id(), sparkline_area.width - 3);
+
+        let histogram_sparkline = MiniHistogram::default()
+            .block(
+                Block::default()
+                    .title("Poll Times Histogram")
+                    .borders(Borders::ALL),
+            )
+            .data(&chart_data)
+            .metadata(metadata)
+            .duration_precision(2);
+
         let task_widget = Paragraph::new(metrics).block(block_for("Task"));
         let wakers_widget = Paragraph::new(vec![wakers, wakeups]).block(block_for("Waker"));
         let fields_widget = Paragraph::new(fields).block(block_for("Fields"));
 
+        let percentiles_widget = block_for("Poll Times Percentiles");
+        let (percentiles_1, percentiles_2) =
+            make_percentiles_widgets(self.details.clone(), task.id());
+        let percentiles_widget_1 = Paragraph::new(percentiles_1);
+        let percentiles_widget_2 = Paragraph::new(percentiles_2);
+
         frame.render_widget(Block::default().title(controls), controls_area);
         frame.render_widget(task_widget, stats_area[0]);
         frame.render_widget(wakers_widget, stats_area[1]);
         frame.render_widget(fields_widget, fields_area);
+        frame.render_widget(percentiles_widget, histogram_area[0]);
+        frame.render_widget(percentiles_widget_1, percentiles_columns[0]);
+        frame.render_widget(percentiles_widget_2, percentiles_columns[1]);
+        frame.render_widget(histogram_sparkline, sparkline_area);
+    }
+}
+
+fn make_chart_data(details: DetailsRef, task_id: u64, width: u16) -> (Vec<u64>, HistogramMetadata) {
```

Comment:
seems like this (and `make_percentiles_widgets`) _could_  be methods on `DetailsRef`...not a blocker though

Reply:
Unfortunately `DetailsRef` is a type alias and the `Details` struct definition lives in `console/src/tasks.rs`. Not sure if we want to move this implementation there? Having an `impl Details {}` here might also be confusing as well.

Reply:
i think it's fine to have an `impl Details` here.

Reply:
Done in https://github.com/tokio-rs/console/pull/47/commits/992f01c561148b35612eaefe05bb9dfaa072f957. Sorry about the back and forth